[libc] Add NULL macro definitions to header files

saturn691 opened the pull request:
By the C standard, <locale.h>, <stddef.h> <stdio.h>, <stdlib.h>, <string.h>, <time.h>, and <wchar.h> require NULL to be defined.

michaelrj-google reviewed (commented):
Thanks for catching this!

saturn691 commented on libc/include/locale.h.def:
```diff
@@ -10,8 +10,8 @@
 #include "__llvm-libc-common.h"
-
 #include "llvm-libc-macros/locale-macros.h"
+#include "llvm-libc-macros/null-macro.h"
```
In the existing `string.h.def` file, NULL is defined in the `.h.def`, should I modify that too?

michaelrj-google replied:
yes, that should be fixed as well.

saturn691 replied:
Fixed in latest revision

saturn691 committed:
[libc] Add NULL macro definitions to header files

By the C standard, <locale.h>, <stddef.h> <stdio.h>, <stdlib.h>,
<string.h>, <time.h>, and <wchar.h> require NULL to be defined.

saturn691 committed:
fixup! [libc] Add NULL macro definitions to header files

saturn691 committed:
fixup! fixup! [libc] Add NULL macro definitions to header files

saturn691 commented:
Hi Tue, I don't have commit access yet, please commit this on behalf of me, thanks!
